### PR TITLE
Create an event and update workload status when failing to schedule a…

### LINKED ISSUE #91: Create an event and update workload status when failing to schedule a workload
Probably for the workloads that wasn't returned in this loop: https://github.com/kubernetes-sigs/kueue/blob/9912f26234f40a9b43f48b5f2cf3abfde15c55bf/pkg/scheduler/scheduler.go#L142

### PR DESCRIPTION
**Pull request by uroy-personal**
… workload

#### Which issue(s) this PR fixes:

Fixes #91

### CODE REVIEW

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -143,24 +143,37 @@ func (s *Scheduler) nominate(ctx context.Context, workloads []workload.Info, sna
...
 	for _, w := range workloads {
+		var skipEntry bool
+		var message string
 		log := log.WithValues("queuedWorkload", klog.KObj(w.Obj), "clusterQueue", klog.KRef("", w.ClusterQueue))
 		cq := snap.ClusterQueues[w.ClusterQueue]
 		if cq == nil {
 			log.V(3).Info("ClusterQueue not found when calculating workload admission requirements")
-			continue
+			skipEntry = true
+			message = "ClusterQueue not found when calculating workload admission requirements"
```
Comment (denkensk) [inline pkg/scheduler/scheduler.go:153]: Consider reusing it with the log to avoid duplication

File: pkg/scheduler/scheduler.go [hunk 2]
```diff
@@ -143,24 +143,37 @@ func (s *Scheduler) nominate(ctx context.Context, workloads []workload.Info, sna
 	log := ctrl.LoggerFrom(ctx)
 	entries := make([]entry, 0, len(workloads))
 	for _, w := range workloads {
+		var skipEntry bool
+		var message string
 		log := log.WithValues("queuedWorkload", klog.KObj(w.Obj), "clusterQueue", klog.KRef("", w.ClusterQueue))
 		cq := snap.ClusterQueues[w.ClusterQueue]
 		if cq == nil {
 			log.V(3).Info("ClusterQueue not found when calculating workload admission requirements")
-			continue
```
Comment (denkensk) [inline pkg/scheduler/scheduler.go:150]: why delete `contine`?

**Review by denkensk — COMMENTED** [review]
I think the failures of scheduling also include `Failed admitting workload by clusterQueue` in admit function.

**Comment by ahg-g** [comment]
Thanks @uroy-personal, we discussed a different approach to this. @ArangoGutierrez will likely followup after https://github.com/kubernetes-sigs/kueue/pull/125 is merged.

The idea is that the event (and status update) will be emitted from the same place; `nominate` should return more information as discussed in https://github.com/kubernetes-sigs/kueue/pull/125#discussion_r830546251

**Comment by uroy-personal** [comment]
> Thanks @uroy-personal, we discussed a different approach to this. @ArangoGutierrez will likely followup after #125 is merged.
> 
> The idea is that the event (and status update) will be emitted from the same place; `nominate` should return more information as discussed in [#125 (comment)](https://github.com/kubernetes-sigs/kueue/pull/125#discussion_r830546251)

Thank you. Looking forward to the merged code :) .. Have a nice day ..

**Comment by ArangoGutierrez** [comment]
https://github.com/kubernetes-sigs/kueue/pull/125 is now merged

**Comment by ahg-g** [comment]
> #125 is now merged

do you want to add the Events as well? [uroy-personal](https://github.com/uroy-personal) perhaps you can look at the PR that @ArangoGutierrez will raise and learn from it how we approach such changes?

**Comment by uroy-personal** [comment]
> > #125 is now merged
> 
> do you want to add the Events as well? [uroy-personal](https://github.com/uroy-personal) perhaps you can look at the PR that @ArangoGutierrez will raise and learn from it how we approach such changes?

Yes sure. I will follow the PR to be raised by @ArangoGutierrez.. Thank you..

**Pull request closed without merging**